tests/serde.rs

```diff
@@ -0,0 +1,95 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+#[cfg(feature = "serde")]
+mod serde {
```

Comment:
you can remove the module and declare that the entire file should only be compiled when the `serde` feature is enabled by adding this on top
```suggestion
#![cfg(feature = "serde")]
```